[Ingest pipelines] Fix bug when deleting pipeline from details panel

alisonelizabeth opened the pull request:
This PR fixes a bug when deleting a pipeline via the details panel. Previously, the details panel did not close and instead displayed the "Pipeline not found" error message after the pipeline was deleted. With this change, the flyout closes and returns to the table view.

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45363/)
* Commit: 92a076ff93701e4b5e97aacb422addb2346bbe9e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45363/","number":"45363","commit":"92a076ff93701e4b5e97aacb422addb2346bbe9e"}]}
PIPELINE-->

alisonelizabeth commented:
> I did find another weird bug though. When the delete confirmation modal appears, the context menu remains open:

@cjcenizal good catch! I fixed this. I also found another bug where if you clicked "Cancel" from the delete modal, the flyout would close. I fixed this as well. Would you mind taking another look?

cjcenizal commented:
@alisonelizabeth 👍 I just tested both interactions and things worked as expected.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45613/)
* Commit: e268da0ca02630ae19029b6a6b579d485eaf8b75

### History
* :green_heart: [Build #45363](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45363/) succeeded 92a076ff93701e4b5e97aacb422addb2346bbe9e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45363/","number":"45363","commit":"92a076ff93701e4b5e97aacb422addb2346bbe9e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/45613/","number":"45613","commit":"e268da0ca02630ae19029b6a6b579d485eaf8b75"}]}
PIPELINE-->
